front/app/components/FilterSelector/MultiSelectDropdown.tsx

```diff
@@ -122,40 +122,68 @@ const MultiSelectDropdown = ({
   isLoading,
 }: Props) => {
   const tabsRef = useRef<(HTMLLIElement | null)[]>([]);
-  const [focusedIndex, setFocusedIndex] = useState(-1);
   const isPhoneOrSmaller = useBreakpoint('phone');
 
   const handleOnToggleCheckbox =
     (entry: IFilterSelectorValue) => (_event: React.ChangeEvent) => {
       onChange(entry.value);
     };
 
-  const handleOnSelectSingleValue =
-    (entry: IFilterSelectorValue) => (event: KeyboardEvent<HTMLLIElement>) => {
-      if (
-        event.type === 'keydown' &&
-        (event.key === 'ArrowUp' || event.key === 'ArrowDown')
-      ) {
-        event.preventDefault();
-        const totalItems = values.length;
-        let nextIndex = 0;
-        if (event.key === 'ArrowUp') {
-          nextIndex = focusedIndex === 0 ? totalItems - 1 : focusedIndex - 1;
-        } else {
-          nextIndex = focusedIndex === totalItems - 1 ? 0 : focusedIndex + 1;
-        }
-        setFocusedIndex(nextIndex);
-      } else if (
-        event.type === 'click' ||
-        (event.type === 'keydown' && event.code === 'Space')
-      ) {
-        event.preventDefault();
-        onChange(entry.value);
+  const handleOnKeyDown = (
+    event:
+      | KeyboardEvent<HTMLLIElement | HTMLButtonElement | HTMLDivElement>
+      | React.MouseEvent<HTMLLIElement>
+      | FormEvent
+  ) => {
+    const target = event.currentTarget as HTMLElement;
+    const indexAttr = target.getAttribute('data-index');
+    const value = target.getAttribute('data-value');
+    const index = indexAttr ? parseInt(indexAttr, 10) : undefined;
+    const keyboardEvent = event as KeyboardEvent<HTMLLIElement>;
+    const key = keyboardEvent.key;
+    const totalItems = values.length;
+
+    if (event.type === 'click') {
+      event.preventDefault();
+      if (value) onChange(value);
+      return;
+    }
+    if (key === 'ArrowDown') {
+      keyboardEvent.preventDefault();
+      if (index === undefined) {
+        if (!opened) toggleValuesList();
+        else handleKeyDown?.(keyboardEvent);
+        return;
       }
-    };
+      // navigate to next item (circular 0,1,2,3,...,0)
+      tabsRef.current[(index + 1) % totalItems]?.focus();
+      return;
+    }
+    if (key === 'ArrowUp') {
+      if (index !== undefined) {
```

Comment:
For the key handling, could we use a `switch` statement here just to make it a bit cleaner (similar to the Combobox)?